api/include/opentelemetry/context/contextvars_context.h

```diff
@@ -0,0 +1,50 @@
+#ifndef CONTEXT_VARS_CONTEXT_H_
+#define CONTEXT_VARS_CONTEXT_H_
+
+#include "context.h"
+#include <string>
+
+
+/* An implementation of the RuntimeContext interface which wraps 
+ * ContextVar under the hood */
+class ContextVarsRuntimeContext: public RuntimeContext{
+
+  std::string _CONTEXT_KEY = "current_context";
+
+
+  /* ContextVarsRuntimeContext: Default constructor to set the 
+   * current context to the thread local context 
+   * 
+   * Args: None
+   */
+  ContextVarsRuntimeContext();
+
+
+  /* attach: Sets the current 'Context' object. Returns a token 
+   * that can be used to reset to the previous Context.
+   * 
+   * Args:
+   *  context : the context to set. 
+   */
+  Context attach(Context context);
+
+
+  /* get_current: Return the current context.
+   *
+   * Args: None 
+   */
+  Context get_current();
+
+  /* detach: Resets the context to a previous value.
+   * 
+   * Args:
+   *  token: A reference to a previous context
+   */
+  Token detach(Token token);
```

Comment:
From the [Detach specification](https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/context/context.md#detach-context) I wouldn't expect a token to be returned.


We'll need to check about what Open Telemetry does for propagating errors (e.g. do they use exceptions, or return a status object).